lib/engine/graph.rb

```diff
@@ -103,6 +106,17 @@ def compute(corporation, routes_only: false)
         end
       end
 
+      if @home_as_token
+        @game.hexes.each do |hex|
+          next unless Array(corporation.coordinates).include?(hex.id)
+
+          hex.tile.city_towns.each do |ct|
+            hex.neighbors.each { |e, _| hexes[hex][e] = true }
+            nodes[ct] = true
+          end
+        end
```

Comment:
Instead of looping over all hexes, does this work?

```suggestion
        hex = @game.hex_by_id(corporation.coordinates)

        hex.tile.city_towns.each do |ct|
          hex.neighbors.each { |e, _| hexes[hex][e] = true }
          nodes[ct] = true
        end
```

Reply:
LGTM other than this

Reply:
Or if the corporation can have multiple homes (I think that's why you'd throw it in an `Array`?), would this work?

```suggestion
        hexes =  Array(corporation.coordinates).map { |h| @game.hex_by_id(h) }
        hexes.each do |hex|
          hex.tile.city_towns.each do |ct|
            hex.neighbors.each { |e, _| hexes[hex][e] = true }
            nodes[ct] = true
          end
        end
```

Reply:
Yes, but I did want to make it flexible enough to handle coordinates being an Array.

Reply:
Yes, I'll use that latter code.

Reply:
Yeah I figured this was minor compared to other Graph perf stuff 🤷‍♂️

Reply:
Oh whoops looks like the line I typed had an extra space after `home_hexes =`

Reply:
Crap. What I get for not running rubocop :-(